File: mobile/lib/models/booking.dart
import 'package:meeting_room_booking/models/room.dart';
import 'package:meeting_room_booking/models/user.dart';

class Booking {
  int? id;
  String? start;
  String? end;
  String? createdAt;
  int? nbPeople;
  User? user;
  Room? room;
  String? status;

  Booking(
      {this.id,
      this.start,
      this.end,
      this.createdAt,
      this.nbPeople,
      this.user,
      this.room,
      this.status});

  Booking.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    start = json['start'];
    end = json['end'];
    createdAt = json['createdAt'];
    nbPeople = json['nbPeople'];
    user = json['user'] != null ? new User.fromJson(json['user']) : null;
    room = json['room'] != null ? new Room.fromJson(json['room']) : null;
    status = json['status'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = id;
    data['start'] = start;
    data['end'] = end;
    data['createdAt'] = createdAt;
    data['nbPeople'] = nbPeople;
    if (user != null) {
      data['user'] = user!.toJson();
    }
    if (room != null) {
      data['room'] = room!.toJson();
    }
    data['status'] = status;
    return data;
  }
}

enum BookingStatus {
  reserved('RESERVED'),
  completed('COMPLETED'),
  cancelled('CANCELLED');

  final String value;

  const BookingStatus(this.value);
}

File: backend/src/main/java/com/paloit/meeting_room_booking/model/Booking.java
package com.paloit.meeting_room_booking.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

import java.io.Serializable;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "booking")
public class Booking implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(name = "start_datetime")
    private LocalDateTime start;
    @Column(name = "end_datetime")
    private LocalDateTime end;
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    private Integer nbPeople;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @NotNull(message = "Status is required")
    private String status;

    public String getStatus() {
        return this.status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public LocalDateTime getStart() {
        return start;
    }

    public void setStart(LocalDateTime start) {
        this.start = start;
    }

    public LocalDateTime getEnd() {
        return end;
    }

    public void setEnd(LocalDateTime end) {
        this.end = end;
    }

    public Integer getNbPeople() {
        return nbPeople;
    }

    public void setNbPeople(Integer nbPeople) {
        this.nbPeople = nbPeople;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }
}
